dropshot/src/router.rs

```diff
@@ -757,6 +976,49 @@ mod test {
         ));
     }
 
+    #[test]
+    #[should_panic(expected = "URI path \"/projects/default\": attempted to \
+                               register route for literal path segment \
+                               \"default\" when a route exists for variable \
+                               path segment (variable name: \"rest\")")]
+    fn test_literal_after_regex() {
+        let mut router = HttpRouter::new();
+        router.insert(new_endpoint(
+            new_handler(),
+            Method::GET,
+            "/projects/{rest:.*}",
+        ));
+        router.insert(new_endpoint(
+            new_handler(),
+            Method::GET,
+            "/projects/default",
+        ));
+    }
+
+    #[test]
+    #[should_panic(expected = "Only the pattern '.*' is currently supported")]
+    fn test_bogus_regex() {
+        let mut router = HttpRouter::new();
+        router.insert(new_endpoint(
+            new_handler(),
+            Method::GET,
+            "/word/{rest:[a-z]*}",
+        ));
+    }
+
+    #[test]
+    #[should_panic(expected = "URI path \"/some/{more:.*}/{stuff}\": \
+                               attempted to match segments after the \
+                               wildcard variable \"more\"")]
+    fn test_after_regex() {
+        let mut router = HttpRouter::new();
+        router.insert(new_endpoint(
+            new_handler(),
+            Method::GET,
+            "/some/{more:.*}/{stuff}",
```

Comment:
Yes, that's what I'd propose. I'll add a test for that; thanks for the suggestion. If that sounds like the wrong approach, please let me know!

Reply:
Actually, we currently treat the presence or absence of that trailing slash as identical. It's an area that has some associated `TODO` comments as I think we need to make a determination one way or the other (see `input_path_to_segments`).

Reply:
Ack. IMO, still worth the test, even with the final state TBD - seems like behavior that we wouldn't want to alter without knowing about it.

Reply:
Fair enough. I'll add a test that succeeds with the final slash and state in the test that it's reasonable to change the behavior although it would be a potentially breaking change.